### PR TITLE
add domain manager support to the client

### PR DESCRIPTION
**Pull request by davidrichardson**

### CODE REVIEW

File: service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java [hunk 1]
```diff
@@ -199,6 +199,34 @@ private Domain aDomain() {
...
+				"  ]\n" +
+				"}";
+
+		this.domainsApi.expect(requestTo(expectedUrl)).andExpect(method(HttpMethod.PUT))
+				.andRespond(withSuccess().body(mockResponse).contentType(MediaType.APPLICATION_JSON));
+
+		Domain updated = subject.addManagerToDomain(toJoin, toAdd, "a-token");
+
+		this.domainsApi.verify();
+		assertThat(updated.getManagers(), notNullValue());
```
Comment (am-cornelis) [inline service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java]: We could make that more explicit with something like `assertThat(updated.getManagers().iterator().next().getUserReference(), equalTo("usr-the-bar"));` (or any other relevant field in the single manager defined in the mockResponse)?

File: service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java [hunk 2]
```diff
@@ -269,6 +297,28 @@ public void can_delete_user_from_domain(){
 		
 		assertThat(toJoin.getUsers().size(), is(0));
 	}
+
+	@Test
+	public void can_delete_manager_from_domain(){
+
+		Domain toJoin = new Domain(null, null, "the-foo");
+		User toAdd = user("the-bar");
```
Comment (am-cornelis) [inline service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java]: rename domain toLeave and user toDelete?

File: service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java [hunk 3]
```diff
@@ -269,6 +297,28 @@ public void can_delete_user_from_domain(){
...
+
+		String expectedUrl = String.format("/domains/dom-%s/managers/usr-%s",  toJoin.getDomainReference(), toAdd.getUserReference());
+
+		this.domainsApi.expect(requestTo(expectedUrl)).andExpect(method(HttpMethod.DELETE))
+				.andRespond(withSuccess().body(mockResponse).contentType(MediaType.APPLICATION_JSON));
+
+		toJoin = subject.removeManagerFromDomain(toAdd, toJoin, "token");
+		this.domainsApi.verify();
+
+		assertThat(toJoin.getManagers().size(), is(0));
```
Comment (am-cornelis) [inline service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java]: I'm not sure about this: it looks like it's checking that `"managers" : [ ]` gets parsed to an empty list (i.e. there's no difference between already empty list and deleting one). 
Would `this.domainsApi.verify()` be sufficient checking (i.e. it'll fail if we don't DELETE on the expected URL)?

File: service/src/integration-test/java/uk/ac/ebi/tsc/aap/client/repo/ApplicationIntegrationTest.java [hunk 4]
```diff
@@ -93,14 +93,33 @@ public void can_add_user_to_a_domain() {
...
+        Domain result = domainService.addManagerToDomain(domain, user, token);
+        assertNotNull(result);
+        Domain deletedResult = domainService.removeManagerFromDomain(user,domain,token);
+        assertNotNull(deletedResult);
+    }
+
     @Test
     public void can_get_all_users_from_domain() {
-        LOGGER.trace("[ApplicationIntegrationTest] - can_add_user_to_a_domain");
+        LOGGER.trace("[ApplicationIntegrationTest] - can_get_all_users_from_a_domain");
```
Comment (am-cornelis) [inline service/src/integration-test/java/uk/ac/ebi/tsc/aap/client/repo/ApplicationIntegrationTest.java]: Thanks!

File: service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java (hunk 1 above)
Comment (davidrichardson) [reply service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java]: Agreed

File: service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java (hunk 2 above)
Comment (davidrichardson) [reply service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java]: Agreed

File: service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java (hunk 3 above)
Comment (davidrichardson) [reply service/src/test/java/uk/ac/ebi/tsc/aap/client/repo/DomainRepositoryRestTest.java]: Agreed, the assert is not useful here

**Comment by am-cornelis** [comment]
Many thanks for this contribution!

The changes have gone through CI, and have been uploaded to oss.sonatype.org. The new artifacts should be available soon.

**Pull request merged**